phantazm-commons/src/main/java/com/github/phantazmnetwork/commons/AdventureConfigProcessors.java

```diff
@@ -0,0 +1,125 @@
+package com.github.phantazmnetwork.commons;
+
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.ConfigPrimitive;
+import com.github.steanky.ethylene.core.collection.ConfigNode;
+import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.key.InvalidKeyException;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.sound.Sound;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public final class AdventureConfigProcessors {
+    private AdventureConfigProcessors() {
+        throw new UnsupportedOperationException();
+    }
+
+    private static final ConfigProcessor<Key> key = new ConfigProcessor<>() {
+        @Override
+        public Key dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+            if(!element.isString()) {
+                throw new ConfigProcessException("Element must be a string");
+            }
+
+            try {
+                //noinspection PatternValidation
+                return Key.key(element.asString());
+            }
+            catch (InvalidKeyException keyException) {
+                throw new ConfigProcessException(keyException);
+            }
+        }
+
+        @Override
+        public @NotNull ConfigElement elementFromData(Key key) {
+            return new ConfigPrimitive(key.asString());
+        }
+    };
+
+    private static class ComponentConfigProcessor implements ConfigProcessor<Component> {
+
+        private final MiniMessage miniMessage;
+
+        private final boolean optional;
+
+        public ComponentConfigProcessor(@NotNull MiniMessage miniMessage, boolean optional) {
+            this.miniMessage = Objects.requireNonNull(miniMessage, "miniMessage");
+            this.optional = optional;
+        }
+
+        @Override
+        public Component dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+            if (optional && element.isNull()) {
```

Comment:
i don't hate this but i would like to know what the expected use case is, because it will likely require broad changes to be made to support nullability across lots of other cfg processors

Reply:
optional display name field

Reply:
can't you just not have the display name field at all if you don't want it to show up?

Reply:
```java
        Component displayName;
        ConfigElement displayNameElement = element.getElementOrDefault(() -> new ConfigPrimitive(null), "displayName");
        if (!element.isNull()) {
            displayName = null;
        }
        else {
            displayName = componentProcessor.dataFromElement(displayNameElement);
        }
```
would look like this, which now idk maybe not terrible